src/lib/components/input/input.js

```diff
@@ -49,9 +49,19 @@ class Input extends PureComponent {
       isValidationOk,
       renderValidationError,
       autoFocus,
-      ...inputProps
+      ...restProps
     } = this.props;
 
+    // removing props that are not valid for DOM elements
+    // to hand the rest to the input
+    const {
+      handleValidations,
```

Comment:
Why do we split props in 2 places? Is it not better to add `handleValidations, onChangeFunction, getValidationMessage, validationMessage, hasErrorAfterSubmit` on `= this.props`? Something like:
```
 const {
      containerClassName,
      className,
      theme,
      placeholder,
      disabled,
      required,
      name,
      value,
      type,
       isValidationOk,
       renderValidationError,
       autoFocus,
       handleValidations,
       onChangeFunction,
       getValidationMessage,
       validationMessage,
       hasErrorAfterSubmit,
       ...inputProps,
     } = this.props;
```